src/test/librbd/test_Groups.cc

```diff
@@ -97,9 +97,15 @@ TEST_F(TestGroup, add_image)
   ASSERT_EQ(0, rbd_get_op_features(image, &op_features));
   ASSERT_TRUE((op_features & RBD_OPERATION_FEATURE_GROUP) == 0ULL);
 
+  rbd_group_info_t group_info;
+  ASSERT_EQ(0, rbd_get_group(image, &group_info, sizeof(group_info)));
```

Comment:
Ah, `rbd_get_group` for an image without group returns no error. But it sets `group_info.pool` to `RBD_GROUP_INVALID_POOL`. So could you please add the additional test:
```
ASSERT_EQ(group_info.pool, RBD_GROUP_INVALID_POOL);
```

And could you please add the same test (before `image.group_image_add`) to `add_imagePP`?